File: rust_exercises/leetcode/Problem718/solution.rs
impl Solution {
    pub fn find_length(a: Vec<i32>, b: Vec<i32>) -> i32 {
        let mut max_len: i32 = 0;
        let mut dp = vec![vec![0i32; b.len() + 1]; a.len() + 1];
        for i in (1..a.len() + 1) {
            for j in (1..b.len() + 1) {
                if a[i - 1] == b[j - 1] {
                    dp[i][j] = dp[i - 1][j - 1] + 1;
                    max_len = std::cmp::max(max_len, dp[i][j]);
                }
            }
        }
        max_len
    }
}

File: rust_exercises/leetcode/Problem718/Solution.java
public class Solution {
    public int findLength(int[] A, int[] B) {
        int[][] dp = new int[A.length + 1][B.length + 1];
        int maxLen = 0;
        for (int i = 1; i <= A.length; i += 1) {
            for (int j = 1; j <= B.length; j += 1) {
                if (A[i - 1] == B[j - 1]) {
                    dp[i][j] = dp[i - 1][j - 1] + 1;
                    maxLen = Math.max(maxLen, dp[i][j]);
                }
            }
        }
        return maxLen;
    }
}
